integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/chat_generator.py

```diff
@@ -0,0 +1,243 @@
+import json
+import logging
+import re
+from typing import Any, Callable, ClassVar, Dict, List, Optional, Type
+
+import boto3
+from botocore.exceptions import BotoCoreError, ClientError
+from haystack import component, default_from_dict, default_to_dict
+from haystack.components.generators.utils import deserialize_callback_handler
+from haystack.dataclasses import ChatMessage, StreamingChunk
+
+from haystack_integrations.components.generators.amazon_bedrock.errors import (
+    AmazonBedrockConfigurationError,
+    AmazonBedrockInferenceError,
+    AWSConfigurationError,
+)
+
+from .adapters import AnthropicClaudeChatAdapter, BedrockModelChatAdapter, MetaLlama2ChatAdapter
+
+logger = logging.getLogger(__name__)
+
+AWS_CONFIGURATION_KEYS = [
+    "aws_access_key_id",
+    "aws_secret_access_key",
+    "aws_session_token",
+    "aws_region_name",
+    "aws_profile_name",
+]
+
+
+@component
+class AmazonBedrockChatGenerator:
+    """
+    AmazonBedrockChatGenerator enables text generation via Amazon Bedrock chat hosted models. For example, to use
+    the Anthropic Claude model, simply initialize the AmazonBedrockChatGenerator with the 'anthropic.claude-v2'
+    model name.
+
+    ```python
+    from haystack_integrations.components.generators.amazon_bedrock import AmazonBedrockChatGenerator
+    from haystack.dataclasses import ChatMessage
+    from haystack.components.generators.utils import print_streaming_chunk
```

Comment:
since this `print_streaming_chunk` is use an example maybe it can be removed

Reply:
But we use it in example?

Reply:
ahh got it, it's on the `streaming_callback`